python/tank/util/shotgun.py

```diff
@@ -215,8 +216,10 @@ def __create_sg_connection(config_data=None):
             # This is needed for backwards compatibility with scripts that were
             # written before authentication was put in place. Since those scripts
             # don't set the current user, we have to get the one configured for the
-            # core instead.
-            user = sa.get_user()
+            # core instead. If the script user is configured, then the
+            # CoreDefaultsManager will provide its credentials for us and
+            # sa.get_default_user will return the ScriptUser instance.
+            user = sa.get_default_user()
```

Comment:
I wonder if we could cache this value at this point... Would that create confusing scenarios? basically just do `set_current_user()`.